### PR TITLE
[WIP] created setting for browser usage

### PR DESCRIPTION
**Pull request by m0veax**
solves #116

I created the settings as you did it for the image quality.

Now I have the problem that i can't access the setting through the Provider in the function itself, because i don't have access to the context object:

https://github.com/jesusrp98/spacex-go/blob/807152bd0111720848f0156749b0fbc42fc19aaa/lib/util/browser.dart#L3

Should i change the function itself to get a second parameter or do you want it to get from the SharedPreferences directly?

### CODE REVIEW

**Comment by jesusrp98** [comment]
Greate job @m0veax !

Ok so what we can do is introduce a new extension on the `BuildContext` class in order to perform this URL actions.
The result would be that opening an URL would be by calling `context.showUrl('http://google.com')`.

You can check more info on extensions [here](https://dart.dev/guides/language/extension-methods).

Also, I've working o a major rework of the app on here (#122 #120).
I'd need to rework this whole implementation after the fact.
Basically, I removed dependencies on `provider` & `shared_preferences` to start using `bloc` & `hydrated_bloc`, a more efficient way to handle & store current app state.

Let me know if you need some help!

**Comment by jesusrp98** [comment]
I'm going to merge the Cubit UI branch in the next couple of hours, which will make this PR incompatible with `master`.
If you want to, I can make this PR compatible with the main branch, and you could make the final touches after that.

What do you think @m0veax ?

**Comment by m0veax** [comment]
That sounds good. I'll wait for your merge and will finish my work afterwards. Thanks for helping out there

**Comment by jesusrp98** [comment]
Since we'd need to update almost everything, I think I'd need to delete all these commits and start over, merging all changes from `master`. I can do that easilly.

We can reuse this branch & PR, delete all previous commits, then I redo all the work you've done, and lastly you can update the `openURL` function, what do you think @m0veax ?

**Comment by m0veax** [comment]
Hey, feel free to go :)

Even if I have to redo something I still want to implement this feature.

Just leave me a note what's need to be done. Btw, I never worked with bloc. I would need a pointer to the docs to solve it in bloc. But I have fun learning something, so just go on :)

**Comment by jesusrp98** [comment]
@m0veax Already deleted your previous work (don't worry about it, I'll reimplement what you've done).
Could you please rebase your forked repo to be in par with the latest `master` commit from the original repo?
I can't do it by myself, look at this [link](https://stackoverflow.com/questions/7244321/how-do-i-update-a-github-forked-repository) about how to do it.

After that, I'd start working into reimplementing this feature :)

**Comment by m0veax** [comment]
@jesusrp98  all done, happy hacking :)

**Comment by jesusrp98** [comment]
@m0veax Been trying to work but i cant create a new branch from your master, nor i can update the `feature-urlhanding` branch to the master.

Could you please do one of those options? :)

**Comment by m0veax** [comment]
Hey,

I created the branch feature-urlhandlingV2 from my master:

https://github.com/m0veax/spacex-go/tree/feature-urlhandlingV2

I hope i did it right. Do you need contributer rights or something else? I could give them to you if needed :)

If you need anything further, just let me now

**Comment by jesusrp98** [comment]
@m0veax yeah having contributer rights to your forked repo would be nice, thanks!

**Comment by m0veax** [comment]
@jesusrp98 I hope that worked out. You should have an invite now
